### PR TITLE
Adding volume to lib_openshift.

### PR DESCRIPTION
**Pull request by kwoodson**

### CODE REVIEW

**Comment by kwoodson** [comment]
aos-ci-test

File: roles/lib_openshift/src/doc/volume [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    default: present
+    choices: ["present", "absent", "list"]
+    aliases: []
+  kubeconfig:
+    description:
+    - The path for the kubeconfig file to use for authentication
+    required: false
+    default: /etc/origin/master/admin.kubeconfig
+    aliases: []
+  debug:
```
Comment (ashcrow) [inline roles/lib_openshift/src/doc/volume]: I didn't see this in the class.

File: roles/lib_openshift/src/doc/volume [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    required: false
+    default: None
+    aliases: []
+  claim_name:
+    description:
+    - The name of the pv claim
+    required: false
+    default: None
+    aliases: []
+author:
```
Comment (ashcrow) [inline roles/lib_openshift/src/doc/volume]: Looks like resource_name, vol_name, and verbose are missing.

File: roles/lib_openshift/src/doc/volume (hunk 1 above)
Comment (kwoodson) [inline roles/lib_openshift/src/doc/volume]: @ashcrow, Its a little difficult to follow but I believe it is here.
https://github.com/kwoodson/openshift-ansible/blob/oc_volume/roles/lib_openshift/src/ansible/oc_volume.py#L14
Then passed into the class here.
https://github.com/kwoodson/openshift-ansible/blob/oc_volume/roles/lib_openshift/src/class/oc_volume.py#L116

**Comment by kwoodson** [comment]
@ewolinetz, please take a look at the configmap code added.

@ashcrow, if you get a few minutes.  I updated this module with the bits to add configmap to volumes.

File: roles/lib_openshift/library/oc_volume.py [hunk 3]
```diff
@@ -1750,24 +1756,28 @@ def create_volume_structure(volume_info):
...
             volume[volume_info['type']] = {'secretName': volume_info['secret_name']}
             volume_mount = {'mountPath': volume_info['path'],
                             'name': volume_info['name']}
-        elif volume_info['type'] == 'emptydir':
+        elif volume_type == 'emptydir':
             volume['emptyDir'] = {}
             volume_mount = {'mountPath': volume_info['path'],
                             'name': volume_info['name']}
-        elif volume_info['type'] == 'pvc':
+        elif volume_type == 'pvc' or volume_type == 'persistentvolumeclaim':
```
Comment (ashcrow) [inline roles/lib_openshift/library/oc_volume.py]: :+1:

File: roles/lib_openshift/library/oc_volume.py [hunk 4]
```diff
@@ -1750,24 +1756,28 @@ def create_volume_structure(volume_info):
         ''' return a properly structured volume '''
         volume_mount = None
         volume = {'name': volume_info['name']}
-        if volume_info['type'] == 'secret':
+        volume_type = volume_info['type'].lower()
```
Comment (ashcrow) [inline roles/lib_openshift/library/oc_volume.py]: :+1: I think switching back to case insensitive is best.

Comment (kwoodson) [inline roles/lib_openshift/library/oc_volume.py]: Yeah, this was my fault.  @mtnbikenc had made changes to this file to do the insensitive case checking.  I stomped his changes in the first PR so I reverted back.  That was my fault.

**Comment by kwoodson** [comment]
@ewolinetz, the failures mentioned by @ashcrow were in the new test I added for configmap.

I have fixed the test and ran `tox` locally and everything looks good.  Let's try this again!

**Comment by ashcrow** [comment]
aos-ci-test
[test]

**Comment by mtnbikenc** [comment]
@ashcrow I've not had success with putting aos-ci-test in the same comment with `bracket-test-bracket`.  So I've just done each in their own comment.

**Comment by mtnbikenc** [comment]
aos-ci-test

**Comment by kwoodson** [comment]
Seems like this flaked out on:
>                 Message: "the server has asked for the client to provide credentials (get deploymentConfigs history-limit)",

Rerunning.

**Comment by kwoodson** [comment]
aos-ci-test

**Comment by kwoodson** [comment]
[test]

**Comment by kwoodson** [comment]
@stevekuznetsov, @sdodson, any ideas on this one?  

I think its unrelated to oc_volume since this is brand new code that is unreferenced.

**Comment by stevekuznetsov** [comment]
@smarterclayton does this look like namespace cleanup failures in serial tests?
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_openshift_ansible_extended_conformance_install/53/#showFailuresLink

e2e framework bug?

**Comment by stevekuznetsov** [comment]
That is one noisy bot ....

**Comment by sdodson** [comment]
[test]

**Comment by kwoodson** [comment]
[merge]

**Comment by smarterclayton** [comment]
@stevekuznetsov yes, if more than one serial test fails it's almost certainly because of namespace cleanup.

**Pull request merged**